arch/arm/Kconfig

```diff
@@ -34,6 +34,67 @@ config ARM_CUSTOM_INTERRUPT_CONTROLLER
 	  is assumed that the custom interrupt control interface implementation
 	  assumes responsibility for handling the NVIC.
 
+config ROMSTART_RELOCATION_ROM
+	bool
+	default n
+	help
+	  Relocates the rom_start region containing the boot-vector data and
+	  irq vectors to the region specified by configurations:
+	  ROMSTART_REGION_ADDRESS and ROMSTART_REGION_SIZE
+
+	  This is useful for the Linux Remoteproc framework that uses the elf-loader
+	  such that it is able to load the correct boot-vector (contained in rom_start)
+	  into the correct memory location independent of the chosen zephyr,flash
+	  ROM region.
+
+	  Most SOCs include an alias for the boot-vector at address 0x00000000
+	  so a default which might be supported by the corresponding Linux rproc driver.
+	  If it is not, this option allows to specify the addresses into the elf file.
+
+	  In general this option should be chosen if the zephyr,flash chosen node
+	  is not placed into the boot-vector memory area
+
+	  Example:
+		on IMX7D, the chosen zephyr,flash can be OCRAM/OCRAM_S/TCM/DDR memories
+		for code location. But the boot-vector must be placed into OCRAM_S for the
+		CORTEX-M to boot (alias 0, real 0x00180000/32K available).
+
+if ROMSTART_RELOCATION_ROM
+
+	config ROMSTART_REGION_ADDRESS
+		hex
+		default 0x00000000
+		help
+		  Start address of the rom_start region.
+		  This setting can be derived from a DT node reg property or specified directly.
+
+		  A default value of 0x00000000 might work in most cases as SOCs have an alias
+		  to the right memory region of the boot-vector.
+
+		  Examples:
+		   IMX7D the boot-vector is OCRAM_S (0x00180000, aliased at 0x0).
+		   IMX6SX the boot-vector is TCML (0x007F8000, aliased at 0x0).
+		   IMX8MQ the boot-vector is TCML (0x007E0000, aliased at 0x0).
+		   IMX8MN the boot-vector is ITCM (0x007E0000, aliased at 0x0).
+
+		  Example of DT definition:
+		  	$(dt_nodelabel_reg_addr_hex,ocram_s_sys)
```

Comment:
Please fix compliance issue firstly:
arch/arm/Kconfig:81: space before tab in indent.

Reply:
Thanks for the review, I will :)

Is there a way to run compliance checks locally before submitting the PR ?
Because I wasn't warned by ./scripts/checkpatch.pl at the time of submission.

Reply:
I think I found the way to check the errors:

```
cd ~/zephyrproject/zephyr
./script/ci/check_compliance.py 
```

Reply:
> I think I found the way to check the errors:
> 
> ```
> cd ~/zephyrproject/zephyr
> ./script/ci/check_compliance.py 
> ```

And pushed the corrections :)